### PR TITLE
ROACHMAP: Pull roaches out of simple_animal

### PR DESCRIPTION
**Pull request by silentenemy**
This commit adds special subtype of `/mob/living/superior_animal` for roaches and spiders.
Roaches and spiders are no more `simple_animal/hostile`, though currently
most procs related to them are copypasted from `simple_animal`.

Structure:

- `superior_animal.dm`
- `ai.dm`
- `defense.dm`
- `life.dm`
- `giant_spider/`
  - `ai.dm`
  - `giant_spider.dm`
- `roach/`
  - `roach.dm`      - Roaches definition; miscellaneous procs.
  - `life.dm`       - The Life And Death of Roach the Fuhrer
  - `defense.dm`    - Damage-related procs, also attack* procs.
  - `ai.dm`         - AI of roaches.
  - `meat.dm`       - Roach meat definition.
  - `blattedin.dm`  - Blattedin-specific reaction; moved from main blattedin code to simplify editing.
  - `types/`
    - `fuhrer.dm`     - All subtypes of roaches
    - `hunter.dm`        have been moved here
    - `support.dm`       to clean out the mess
    - `tank.dm`          with subtype-specific procs.

### CODE REVIEW

**Comment by NanakoAC** [comment]
superior animal?
Why not complex animal, if such a thing is needed? What's the plan here?

One of my future porting plans involves adding reagent and nutrition support to simple animals, i'm concerned about a potential conflict of design

File: code/modules/mob/living/superior_animal/ai.dm [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/mob/living/superior_animal
+	var/stance = HOSTILE_STANCE_IDLE	//Used to determine behavior
```
Comment (NanakoAC) [inline code/modules/mob/living/superior_animal/ai.dm]: I'd say variables should be defined in superior_animal.dm, all of them

File: code/modules/mob/living/superior_animal/giant_spider/ai.dm [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+	. = ..()
+	if(!.)
+		walk(src, 0)
+		return 0
+	if(client)
+		return 0
+
+	if(!stat)
+		switch(stance)
```
Comment (NanakoAC) [inline code/modules/mob/living/superior_animal/giant_spider/ai.dm]: Maybe look at moving this AI stuff out of life and into a better system ?

**Comment by NanakoAC** [comment]
a thought as well, why not move superior animals up to mob/living/carbon. There's more features they could potentially take advantage of

right now i dont see any gain in having a branch at the same level as simple animal, and you will inevitably start duplicating code from farther along the tree if you make them more complex

**Comment by silentenemy** [comment]
>What's the need to make them seperate? Why not improve the hostile animal class instead? or even extend it like simple_animal/hostile/superior.

There are lots of features planned (#1468) which require rewrite of simple_animal procs. On the other hand carbon subtype brings some things (such as viruses) that are unnecessary for roaches, spiders and so.

**Pull request merged**